### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by bernabasy**
### RUBY / DATABASES AND INTERVIEWING: OOP school library/set up associations
**_In this pull request I have:_**
- Created a class Classroom.
- Created the has-many/belongs-to relationship between Classroom and Student.
- Created a class Book.
- Created a class Rental.
- Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental.

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label, :students
```
Comment (cynthiainga) [inline classroom.rb]: - [x] The `student` should be an `attr_reader`, please kindly fix that 🙃 🙃

File: student.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 require './person'
 
 class Student < person
+  attr_accessor :classroom
```
Comment (cynthiainga) [inline student.rb]: - [x] The `classroom` should be an `attr_reader`, please kindly fix that 😁 😁

File: student.rb [hunk 3]
```diff
@@ -1,6 +1,8 @@
...
+
```
Comment (cynthiainga) [inline student.rb:5]: - [x] As required in the project you should create a method called `classroom` in the student file that serves as the 'belongs-to' side (a student belongs to a classroom), please kindly fix that 🙃 😁. You can read this [article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships) for more information about the association 😉 

Screen Shot 2023-03-15 at 14 45 10

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_reader :rentals
```
Comment (cynthiainga) [inline book.rb]: - [x] The `rentals` should be `Getters and Setters` which means it should be an `attr_accessor`, please kindly fix that 😉 🙃

File: rental.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_reader :person, :book
```
Comment (cynthiainga) [inline rental.rb]: - [x] The `person` and `book` should also be the `attr_accessor`, kindly fix that as well 😉 😉

File: person.rb [hunk 6]
```diff
@@ -15,7 +16,11 @@ def correct_name
 
   attr_reader :id
 
-  attr_accessor :name, :age
+  attr_accessor :name, :age, :rentals
+
+  def add_rentals(date, book)
+    @Rental.new(date, self, book)
```
Comment (cynthiainga) [inline person.rb]: - [x] Please kindly delete the `@` before the `Rental`, `Rental` is a class and not a variable 😁 😉

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @bernabasy, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] No linters errors ✔️ 
- [ ] Descriptive PR ✔️ 
- [ ] Good commit history ✔️ 
- [ ] Correct Gitflow ✔️ 

## Required Changes ♻️

- [x] Please kindly check the comments under the review 👇 👇 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by brytebee — APPROVED** [review]
Hi, @bernabasy👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great DB design.✔️
- App functions as it should.✔️
- Great implementation of suggested changes adding all attributes.✔️

##

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



Good luck! 🎉 

##

###  **Optional Suggestions:**
- N/A

------

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**